### PR TITLE
Rename DeckTask to CollectionTask

### PR DESCRIPTION
**Pull request by Arthur-Milchior**
This class was created in 2009. If I understand code history correctly
"Deck" has been renamed to "Collection" and what is called "deck"
today is totally distinct.

This commit was done using android studio, so I am quite confident it
did not miss anything. And I hope it'll make the code slightly less
confusing for newcomers. (I actually lost 5 minutes trying to figure
out where to put an async task for a non deck-related task)

### CODE REVIEW

File: AnkiDroid/src/main/java/com/ichi2/async/CollectionTask.java [hunk 1]
```diff
@@ -69,7 +69,7 @@
 /**
  * Loading in the background, so that AnkiDroid does not look like frozen.
  */
-public class DeckTask extends BaseAsyncTask<DeckTask.TaskData, DeckTask.TaskData, DeckTask.TaskData> {
+public class CollectionTask extends BaseAsyncTask<CollectionTask.TaskData, CollectionTask.TaskData, CollectionTask.TaskData> {
```
Comment (timrae) [inline AnkiDroid/src/main/java/com/ichi2/async/CollectionTask.java:72]: Codacy Issue found: [Avoid really long classes.](https://app.codacy.com/manual/timrae/Anki-Android/pullRequest?prid=5345393)

**Comment by timrae** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
+ Solved 6
- Added 2
           

Complexity increasing per file
==============================
- AnkiDroid/src/main/java/com/ichi2/async/CollectionTask.java  45
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/timrae/Anki-Android/pullRequest?prid=5345393&bid=17501034)

File: AnkiDroid/src/main/java/com/ichi2/async/CollectionTask.java [hunk 2]
```diff
@@ -197,10 +197,10 @@ public static void cancelTask(int taskType) {
 
     private final int mType;
     private final Listener mListener;
-    private DeckTask mPreviousTask;
+    private CollectionTask mPreviousTask;
```
Comment (timrae) [inline AnkiDroid/src/main/java/com/ichi2/async/CollectionTask.java:200]: Codacy Issue found: [Fields should be declared at the top of the class, before any method declarations, constructors, initializers or inner classes.](https://app.codacy.com/manual/timrae/Anki-Android/pullRequest?prid=5345393)

**Comment by mikehardy** [comment]
out as 2.10alpha68 as soon as google is done processing it...

**Comment by mikehardy** [comment]
Released as 2.10alpha69 as soon as google finishes processing it

**Pull request merged**